docs/template_plugin/tests/functional/op_reference/region_yolo.cpp

```diff
@@ -0,0 +1,194 @@
+        type_converted_data.resize(size / sizeof(float));
+        inputs_fs.read(reinterpret_cast<char*>(file_content.data()), size);
+
+        std::transform(file_content.begin(), file_content.end(), type_converted_data.begin(), [](float x) { return (T)x;});
+        return type_converted_data;
+    }
+};
+
+class ReferenceRegionYoloLayerTest : public testing::TestWithParam<RegionYoloParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params);
+        inputData = {params.inputData};
+        refOutData = {params.refData};
+    }
+    static std::string getTestCaseName(const testing::TestParamInfo<RegionYoloParams>& obj) {
+        auto param = obj.param;
+        std::ostringstream result;
+        result << "inputShape=" << param.inputShape << "_";
+        result << "iType=" << param.inType << "_";
+        result << "oType=" << param.outType << "_";
+        result << "coords=" << param.coords << "_";
+        result << "classes=" << param.classes << "_";
+        result << "num=" << param.num << "_";
+        result << "do_softmax=" << param.do_softmax << "_";
+        result << "mask=" << param.mask << "_";
+        result << "axis=" << param.axis << "_";
+        result << "end_axis=" << param.end_axis;
+        if (param.testcaseName != "")
+            result << "_" << param.testcaseName;
+        return result.str();
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(const RegionYoloParams& params) {
+        const auto p = std::make_shared<op::v0::Parameter>(params.inType, params.inputShape);
+        const auto RegionYolo = std::make_shared<op::v0::RegionYolo>(p,
+                                                                     params.coords,
+                                                                     params.classes,
+                                                                     params.num,
+                                                                     params.do_softmax,
+                                                                     params.mask,
+                                                                     params.axis,
+                                                                     params.end_axis);
+        return std::make_shared<ov::Function>(NodeVector {RegionYolo}, ParameterVector {p});
+    }
+};
+
+TEST_P(ReferenceRegionYoloLayerTest, CompareWithRefs) {
+    Exec();
+}
+
+template <element::Type_t IN_ET>
+std::vector<RegionYoloParams> generateRegionYoloParams() {
+    using T = typename element_type_traits<IN_ET>::value_type;
+
+    std::vector<RegionYoloParams> regionYoloParams {
+        RegionYoloParams(5, 4, 20, true, 1, 3,
+                        1, 125, 13, 13,
+                        std::vector<int64_t>{0, 1, 2},
+                        IN_ET,
+                        RegionYoloParams::read_binary_file<T>("/region_in_yolov2_caffe.data"),
```

Comment:
Why cannot we generate this file in the code?

Reply:
My understanding is that we need to use the test files from ngraph backend test. Do you know any history of those test files for RegionYolo ngrpha backend test?

Reply:
@ilyachur @iefode Original test files are located in src/core/tests/files(https://github.com/openvinotoolkit/openvino/blob/d9acc137f8af41d0bb5aa111abb2dc6bc340fd82/src/core/tests/files/region_in_yolov2_caffe.data).
openvino/cmake/test_model_zoo.cmake seems to copy those files into runtime output directory.

Reply:
Can we use those test data file in current test_model_zoo?

Reply:
templateFuncTests in azure seems not to access to test_model_zoo while unit-test can.

![image](https://user-images.githubusercontent.com/1720339/144343126-cfe096dc-a6c9-4821-aaff-315da467b999.png)

Reply:
Maybe work directories are different

Reply:
@ilyachur @iefode  Do you have any idea why unit-test/other functional test which are using ./test_model_zoo are ok?
I added add_dependency(${TARGET_NAME}, test_model_zoo) but still failed in azure CI.

Reply:
@iefode Still this PR is blocked because CI of ov_template_func_test cannot refer ./test_model_zoo. Do I add those thest cases using files from test_model_zoo in skip_config_test and merge PR first?

Reply:
Target test files are on right place from build as below .
![image](https://user-images.githubusercontent.com/1720339/150069713-26ca8f4f-3cb4-4c81-8b7a-f2fc5b1a2cb2.png)

But VM of ov_template_func_test seems not to have those files on place as below. I think VM configuration doesn't have a module of importing test_model_zoo.
![image](https://user-images.githubusercontent.com/1720339/150069999-fd75e72c-8fd9-4da6-9732-a1afa510e60f.png)

I will remove the test cases,